db/schema.go

```
@@ -155,48 +156,190 @@ func createTriggerFunc(tx *pgx.Tx) error {
-		rows, err := conn.Query(`
-		SELECT tablename FROM pg_catalog.pg_tables WHERE schemaname = $1`, schema,
-		)
-		if err != nil {
-			return nil, err
+	if len(includeTables) > 0 {
+		for _, table := range includeTables {
+			tableRegister[table] = true
 		}
-
-		for rows.Next() {
-			var t string
-			err = rows.Scan(&t)
+	} else {
+		for _, schema := range schemas {
+			rows, err := conn.Query(`
+		SELECT tablename FROM pg_catalog.pg_tables WHERE schemaname = $1`, schema,
+			)
 			if err != nil {
-				return nil, nil
+				return nil, err
 			}
 
-			if _, ok := exclude[t]; !ok {
-				tables = append(tables, fmt.Sprintf("%s.%s", schema, t))
+			for rows.Next() {
+				var t string
+				err = rows.Scan(&t)
+				if err != nil {
+					return nil, err
+				}
+
+				table := fmt.Sprintf("%s.%s", schema, t)
+				tableRegister[table] = true
 			}
 		}
 	}
 
+	for _, table := range excludeTables {
+		if _, ok := tableRegister[table]; ok {
+			tableRegister[table] = false
+		}
+	}
+
+	tables := make([]Table, 0)
+	for tableName, include := range tableRegister {
+		if include {
+			tableDetails, err := getTableDetails(conn, tableName)
+			if err != nil {
+				return nil, err
+			}
+			tables = append(tables, *tableDetails)
+		}
+	}
+
 	return tables, nil
 }
 
-func registerTrigger(tx *pgx.Tx, table string) error {
-	// trigger name is <schema>__<table>_changesets
-	triggerName := strings.ReplaceAll(table, ".", "__")
-	_, err := tx.Exec(fmt.Sprintf(`
-		CREATE TRIGGER %s_changesets
-		AFTER
-			INSERT OR UPDATE OR DELETE
-		ON %s
-		FOR EACH ROW EXECUTE PROCEDURE warp_pipe.on_modify()`, triggerName, table),
+func getTableDetails(conn *pgx.Conn, name string) (*Table, error) {
+	var table Table
+	nameArr := strings.SplitN(name, ".", 2)
+	if len(nameArr) > 1 {
+		table.Schema = nameArr[0]
+		table.Name = nameArr[1]
+	} else {
+		table.Schema = "public"
+		table.Name = nameArr[0]
+	}
+	table.PKeyFields = make(map[int]string)
+	rows, err := conn.Query(`
```

This error is not handled.